exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -388,16 +313,18 @@ public void setupHashTable() throws IOException, SchemaChangeException, ClassTra
    *  Call only after num partitions is known
    */
   private void delayedSetup() {
-
+    setup = true;
```

Comment:
Should be moved down to the end of the method (e.g., in case we throw in the middle) - also logically setup is done at the end.